packages/wrangler/src/deploy/deploy.ts

```diff
@@ -168,6 +168,53 @@ function errIsStartupErr(err: unknown): err is ParseError & { code: 10021 } {
 	return false;
 }
 
+export const validateRoutes = (
+	routes: Route[],
+	hasExperimentalAssets: boolean
+) => {
+	const invalidRoutes: { [route: string]: string[] } = {};
+	for (const route of routes) {
+		if (typeof route !== "string" && route.custom_domain) {
+			if (route.pattern.includes("*")) {
+				invalidRoutes[route.pattern] ??= [];
+				invalidRoutes[route.pattern].push(
+					`Wildcard operators (*) are not allowed in Custom Domains`
+				);
+			}
+			if (route.pattern.includes("/")) {
+				invalidRoutes[route.pattern] ??= [];
+				invalidRoutes[route.pattern].push(
+					`Paths are not allowed in Custom Domains`
+				);
+			}
+		} else if (hasExperimentalAssets) {
```

Comment:
I'm wondering...if the purpose of this function to surface all route validation errors? Ideally IMHO we should aim for that, because surfacing these one at a time is rather frustrating. To that end, should this be an `if..else` block, or rather a series of `if` blocks?

Reply:
This should be surfacing everything at once already (see the tests) - unless i've misunderstood this comment...? 😅

Reply:
oh yeah you're right. gotta love tests. carry on 😋